backend/src/test/java/com/google/models/TimestampTest.java

```diff
@@ -0,0 +1,73 @@
+package com.google.models;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import java.time.format.DateTimeParseException;
+
+
+/** Contain tests for methods in {@link Timestamp} class. */
+@RunWith(JUnit4.class)
+public final class TimestampTest {
+  private static final int DAY_CONST = 1;
+  private static final int MONTH_CONST = 1;
+  private static final int YEAR_CONST = 2000;
+  private static final Timestamp timestamp = new Timestamp(DAY_CONST, MONTH_CONST, YEAR_CONST);
+
+  @Test
+  public void constructor_convertStringToDate() {
+    Timestamp stringTimestamp = new Timestamp("2020-12-5");
+    Timestamp stringTimestampPaddedZero = new Timestamp("2020-12-05");
+    Timestamp stringTimestampDiff = new Timestamp("2020-1-01");
+
+    assertEquals(stringTimestamp.getDay(), 5);
+    assertEquals(stringTimestamp.getMonth(), 12);
+    assertEquals(stringTimestamp.getYear(), 2020);
+    assertEquals(stringTimestampPaddedZero.getDay(), 5);
+    assertEquals(stringTimestampPaddedZero.getMonth(), 12);
+    assertEquals(stringTimestampPaddedZero.getYear(), 2020);
+    assertEquals(stringTimestampDiff.getDay(), 1);
+    assertEquals(stringTimestampDiff.getMonth(), 1);
+    assertEquals(stringTimestampDiff.getYear(), 2020);
+  }
+
+  @Test(expected = DateTimeParseException.class)
+  public void constructor_throwsExceptionForBadFormat_1() {
+    Timestamp stringTimestamp = new Timestamp("202012-5");
+  }
+
+  @Test(expected = DateTimeParseException.class)
+  public void constructor_throwsExceptionForBadFormat_2() {
+    Timestamp stringTimestamp = new Timestamp("2020-13-5");
+  }
+
+  @Test(expected = DateTimeParseException.class)
+  public void constructor_throwsExceptionForBadFormat_3() {
+    Timestamp stringTimestamp = new Timestamp("2020-13-1-5");
+  }
```

Comment:
v nit/optional: i think just generally suffixing with the _1, _2, _3 is probably not the best practice, but it's not a hard requirement from me.

Reply:
optional: you could rename them like "missing dash", "invalid month", "extra dash"

Reply:
nice suggestion @cyu6 :) 

i will use that pattern